api/envoy/extensions/wasm/v3/wasm.proto

```diff
@@ -105,6 +105,10 @@ message VmConfig {
   // vars just like when you do on native platforms.
   // Warning: Envoy rejects the configuration if there's conflict of key space.
   EnvironmentVariables environment_variables = 7;
+
+  // Specifies the number of restarting this VM when it fails due to runtime errors, or bugs.
```

Comment:
Looking at https://github.com/envoyproxy/envoy/issues/16726, the rationale seems to be https://github.com/proxy-wasm/spec/blob/master/docs/WebAssembly-in-Envoy.md#fault-detection-and-notification. Is the idea really to restart and hope-and-pray the same Wasm code does the right thing, or is this to somehow deal with cross-contamination of one tenant in a VM from another?

Seems super fragile TBH, @PiotrSikora you were confident this is a must have, what's the reasoning?

Reply:
This isn't about cross-contamination, since tenants in multi-tenant enviroments generally use different WasmVM instances.

The reasoning behind this feature is resiliance against crashing edge-cases (e.g. Wasm plugin might crash upon receiving binary header values). If the Wasm plugin receives such Query-of-Death once in a while and it crashes, then we could either blackhole all the traffic until a fix is created and rolled-out, or we could automatically restart WasmVM affecting negilible portion of the total traffic.

Note that the crash-invoking request is already terminated upon crash and it won't be processed again (...unless the downstream retries).

Of course, this isn't going to help with a Wasm plugin that's completely broken and crashes on each request, but that's why we have the counter here (I think we should use a rate-limit instead, btw, @mathetake).

Reply:
Makes sense I think. API LGTM modulo some doc nit.